content/en/blog/2024/ambient-vs-cilium/index.md

```diff
@@ -0,0 +1,67 @@
+keywords: [istio,cilium,analysis]
+---
+
+A common question from prospective Istio users is "how does Istio compare to Cilium?"  While Cilium originally only provided L3/L4 Network Policy, recent releases have added service mesh functionality with Envoy, as well as Wireguard Encryption. Like Istio, Cilium is a CNCF Graduated project, and has been around in the community for many years.
+
+Despite offering a similar feature set on the surface, the two projects have substantially different architectures, most notably cilium’s eBPF program for processing L4 in the kernel, contrasted with Istio’s ztunnel component for L4 in user space. These differences have resulted in substantial speculation about how Istio will perform at scale compared to Cilium.
+
+Rather than emphasizing theoretical performance, we put Istio Ambient and Cilium through their paces, focusing on key metrics like latency, throughput, and resource consumption. We also cranked up the pressure with realistic load scenarios, simulating a bustling Kubernetes environment. Finally, we pushed the size of our AKS cluster up to 1000 nodes on 11,000 cores, to understand how these projects perform at scale. Our results show areas where each can improve, but also indicate that Istio is the clear winner at scale.
+
+## Test Scenario
+
+In order to push Istio and Cilium to their limits, we created 500 different services, each backed by 100 pods. Each service is in a separate namespace, which also contains one Fortio load generator client. We restricted the clients to a node pool of 100 32 core machines, to eliminate noise from collocated clients, and allocated the remaining 900 8 core instances to our services.
+
+{{< image width="60%"
+    link="./scale-scenario.png"
+    alt="Scaling to 500 services with 50,000 pods."
+    >}}
+
+For the Istio test, we used Istio’s Ambient Mode, with a Waypoint in every service namespace, and default install parameters. In order to make our test scenarios similar, we had to turn on a few non-default features in Cilium, including Wireguard Encryption, L7 Proxies, and Node Init. We also created a Cilium Network Policy in each namespace, with HTTP path-based rules. In both scenarios, we generated churn by scaling one service to between 85 and 115 instances at random every second, and relabeling one namespace every minute. To see the precise settings we used, and to reproduce our results, see this site.
+
+## Scalability Scorecard
+
+{{< image width="80%"
+    link="./scale-scorecard.png"
+    alt="Scalability Scorecard: Istio vs. Cilium!"
+    >}}
+
+* **The Cilium Slowdown** Cilium, while boasting impressive low latency with default install parameters, slows down substantially when Istio’s baseline features such as L7 policy and encryption are turned on. Additionally, cilium’s memory and CPU utilization remained high even when no traffic was flowing in the mesh. This can impact the overall stability and reliability of your cluster, especially as it grows.
```

Comment:
> while boasting impressive low latency with default install parameters

Given we don't show any data, test setup, explanation, etc of this, it feels out of place

Reply:
If you're referring to the comment that cilium is fast with default params, I think this is generally assumed, and I do have test data to support it.  It's not the focus of this blog post, but I think it's important for fairness to note that Cilium is quite fast when not using encryption and L7...